src/query/graphite/native/builtin_functions.go

```diff
@@ -863,16 +866,20 @@ func movingAverage(ctx *common.Context, input singlePathSpec, windowSizeValue ge
 						if !math.IsNaN(prev) {
 							sum -= prev
 							num--
+						} else {
+							nans--
 						}
 					}
 					next := bootstrap.ValueAt(i + offset - 1)
 					if !math.IsNaN(next) {
 						sum += next
 						num++
+					} else {
+						nans++
 					}
 				}
 
-				if num > 0 {
+				if num > 0 && float64(windowPoints - nans / (windowPoints)) >= xFilesFactor  {
```

Comment:
I believe this is wrong since it's doing integer division and also `nans / (windowPoints)` occurs before the subtraction. Example here (curr vs proposed):
https://play.golang.org/p/upXNlEygWVH

Can you create an `effectiveXFF` function so this is done once right everywhere? 

Proposed:
```go
func effectiveXFF(windowPoints, nans int, xFilesFactor float64) bool {
	return float64(windowPoints - nans) / float64(windowPoints) >= xFilesFactor
}
```

Reply:
Also can we make this `nans < windowPoints` to match the other places this condition is calculated?